scrapper.py

```diff
@@ -31,67 +44,149 @@ class Crawler:
         """
         Finds articles
         """
-        pass
+        self.get_search_urls()
+        for url in self.seed_urls:
+            response = requests.get(url, headers=HEADERS)
+            if not response:
+                raise IncorrectURLError
+            page_soup = BeautifulSoup(response.content, features='lxml')
+            main_soup = page_soup.find('main', id='main')
+            articles_soup = main_soup.find_all('article')
+            for i in range(self.max_articles_per_seed):
+                if len(self.urls) < self.max_articles:
+                    self.urls.append(self._extract_url(articles_soup[i]))
+                else:
+                    break
+            if len(self.urls) == self.max_articles:
+                break
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        default_url = self.seed_urls[0]
+        for i in range(2, 13):
+            self.seed_urls.append(f'{default_url}/page/{i}')
 
 
 class ArticleParser:
     """
     ArticleParser implementation
     """
     def __init__(self, full_url: str, article_id: int):
-        pass
+        self.full_url = full_url
+        self.article_id = article_id
+        self.article = Article(full_url, article_id)
 
     def _fill_article_with_text(self, article_soup):
-        pass
+        article_texts = article_soup.find_all('p')
+        for par in article_texts:
+            if 'class' not in par.attrs:
+                self.article.text += par.text.strip() + ' '
 
     def _fill_article_with_meta_information(self, article_soup):
-        pass
+        # find title
+        self.article.title = article_soup.find('h1', class_="entry-title").text
+
+        # find topics
+        for topic in article_soup.find_all('a', rel="category tag"):
+            self.article.topics.append(topic.text)
+
+        # find author
+        self.article.author = article_soup.find('span', class_="author vcard").find('a').text
+
+        # find date
+        date_art = self.unify_date_format(article_soup.find('time', class_="entry-date published").text)
+        self.article.date = date_art
 
     @staticmethod
     def unify_date_format(date_str):
         """
-        Unifies date format
+        Unifies date format: strftime("%Y-%m-%d %H:%M:%S")
         """
-        pass
+        date = '-'.join(date_str.split('.')[::-1])
+        date_time = str(datetime.strptime(date, "%Y-%m-%d"))
+
+        return datetime.strptime(date_time, "%Y-%m-%d %H:%M:%S")
 
     def parse(self):
         """
         Parses each article
         """
-        pass
+        response = requests.get(self.full_url, headers=HEADERS)
+        if not response:
+            raise IncorrectURLError
+
+        article_soup = BeautifulSoup(response.content, features='lxml')
+        self._fill_article_with_text(article_soup)
+        self._fill_article_with_meta_information(article_soup)
+        self.article.save_raw()
```

Comment:
the method interface is conflicting with requirements.

Reply:
i don't understand with which requirements?